File: app/src/main/java/com/example/mysecondapp/MainActivity.java
package com.example.mysecondapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        EditText editTextNumber1 = findViewById(R.id.editTextNumber1);
        EditText editTextNumber2 = findViewById(R.id.editTextNumber2);
        Button button = findViewById(R.id.button);
        TextView textView = findViewById(R.id.textView);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String str1 = String.valueOf(editTextNumber1.getText());
                int val1 = Integer.parseInt(str1);
                String str2 = String.valueOf(editTextNumber2.getText());
                int val2 = Integer.parseInt(str2);
                textView.setText(Integer.toString(val1 + val2));
            }
        });
    }
}

File: app/src/main/java/com/example/app/MainActivity.kt
package com.example.app

import android.content.Context
import android.content.Intent
import android.hardware.Sensor
import android.hardware.SensorEvent
import android.hardware.SensorEventListener
import android.hardware.SensorManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import androidx.core.text.set
import com.example.app.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity(), SensorEventListener {

    lateinit var binding : ActivityMainBinding
    private lateinit var sensorManager: SensorManager
    private var mAccel : Sensor? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.editText1.setText("0.0")
        binding.editText2.setText("0.0")
        binding.editText3.setText("0.0")
        sensorManager = getSystemService(Context.SENSOR_SERVICE) as SensorManager
        mAccel = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER)
    }

    override fun onSensorChanged(event: SensorEvent?) {
        if (event?.sensor?.type == Sensor.TYPE_ACCELEROMETER) {
            var x = event.values[0]
            var y = event.values[1]
            var z = event.values[2]
            binding.editText1.setText(x.toString())
            binding.editText2.setText(y.toString())
            binding.editText3.setText(z.toString())
            if (x > 20.0 || y > 20.0 || z > 20.0) {
                var intent = Intent(this, DisplayActivity::class.java)
                startActivity(intent)
            }
        }
    }

    override fun onAccuracyChanged(p0: Sensor?, p1: Int) {}

    override fun onResume() {
        super.onResume()
        mAccel?.also { light ->
            sensorManager.registerListener(this, light, SensorManager.SENSOR_DELAY_NORMAL)
        }
    }

    override fun onPause() {
        super.onPause()
        sensorManager.unregisterListener(this)
    }

}
